### PR TITLE
.*: Verify if deletion marker file exists before using it.

### PR DESCRIPTION
**Pull request by pstibrany**
This is an alternative to #2365. Please see the problem description there. This PR modifies `ReadDeletionMark` function to use `Exists` call before reading deletion marker file. This helps to keep `thanos_objstore_bucket_operation_failures_total` down, since missing deletion marker file is not really a failure.

* [x] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

### CODE REVIEW

File: pkg/block/metadata/deletionmark.go [hunk 1]
```diff
@@ -48,6 +48,17 @@ type DeletionMark struct {
 func ReadDeletionMark(ctx context.Context, bkt objstore.BucketReader, logger log.Logger, dir string) (*DeletionMark, error) {
 	deletionMarkFile := path.Join(dir, DeletionMarkFilename)
 
+	// BucketReader.Get reports missing file as a failure (via metrics),
+	// but since most blocks don't have this marker file, it skews metrics unnecessarily.
+	// By using exists first, we avoid that.
+	exists, err := bkt.Exists(ctx, deletionMarkFile)
```
Comment (pracucci) [inline pkg/block/metadata/deletionmark.go:54]: I'm dubious about having to do another API call just to check if the object exists, when we can get the same information from `Get()`. I personally believe #2365 works better ([my take here](https://github.com/thanos-io/thanos/pull/2365#issuecomment-608260426)).

**Comment by bwplotka** [comment]
See discussion here: https://github.com/thanos-io/thanos/pull/2365 and on slack https://cloud-native.slack.com/archives/CL25937SP/p1585903230146500

File: pkg/block/metadata/deletionmark.go (hunk 1 above)
Comment (bwplotka) [reply pkg/block/metadata/deletionmark.go:54]: ack

**Comment by bwplotka** [comment]
Alternative we agrred offline for: https://github.com/thanos-io/thanos/pull/2370 Let me know if that's make sense :hugs:

**Comment by bwplotka** [comment]
Fix merged https://github.com/thanos-io/thanos/pull/2383

**Comment by bwplotka** [comment]
Thanks for help on this :heart:

**Pull request closed without merging**